Emby.Server.Implementations/Playlists/PlaylistManager.cs

```diff
@@ -537,5 +529,53 @@ public Folder GetPlaylistsFolder(Guid userId)
             return _libraryManager.RootFolder.Children.OfType<Folder>().FirstOrDefault(i => string.Equals(i.GetType().Name, TypeName, StringComparison.Ordinal)) ??
                 _libraryManager.GetUserRootFolder().Children.OfType<Folder>().FirstOrDefault(i => string.Equals(i.GetType().Name, TypeName, StringComparison.Ordinal));
         }
+
+        public async Task RemovePlaylists(Guid userId)
+        {
+            var playlists = GetPlaylists(userId);
+            foreach (var playlist in playlists)
+            {
+                // Update owner if shared
+                var rankedShares = playlist.Shares.OrderByDescending(x => x.CanEdit).ToArray();
+                if (rankedShares.Length > 0 && Guid.TryParse(rankedShares[0].UserId, out var guid))
+                {
+                    playlist.OwnerUserId = guid;
+                    playlist.Shares = rankedShares.Skip(1).ToArray();
+                    await playlist.UpdateToRepositoryAsync(ItemUpdateType.MetadataEdit, CancellationToken.None).ConfigureAwait(false);
+
+                    if (playlist.IsFile)
+                    {
+                        SavePlaylistFile(playlist);
+                    }
+                }
+                else
+                {
+                    // Remove playlist if not shared
+                    _libraryManager.DeleteItem(
+                        playlist,
+                        new DeleteOptions
+                        {
+                            DeleteFileLocation = false,
+                            DeleteFromExternalProvider = false
+                        },
+                        playlist.GetParent(),
+                        false);
+                }
+            }
+        }
+
+        public async Task UpdatePlaylist(Playlist playlist)
+        {
+            var currentPlaylist = (Playlist)_libraryManager.GetItemById(playlist.Id);
+            currentPlaylist.OwnerUserId = playlist.OwnerUserId;
+            currentPlaylist.Shares = playlist.Shares;
+
+            await playlist.UpdateToRepositoryAsync(ItemUpdateType.MetadataEdit, CancellationToken.None).ConfigureAwait(false);
+
+            if (playlist.IsFile)
+            {
+                SavePlaylistFile(currentPlaylist);
```

Comment:
Shouldn't the updated `playlist` be saved to file instead of `currentPlaylist`, or am I missing something here?

Reply:
The idea is to merge the modified data from `playlist` into `currentPlaylist` in case `playlist` is incremental and save `currentPlaylist` aferwards.